src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelationClientOptions.cs

```diff
@@ -39,7 +42,11 @@ public string UpstreamServiceHeaderName
             get => _upstreamServiceHeaderName;
             set
             {
-                Guard.NotNullOrWhitespace(value, nameof(value), "Requires a non-blank value for the HTTP request header where the dependency ID should be added when tracking HTTP dependencies");
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new ArgumentNullException(nameof(value), "Requires a non-blank value for the HTTP request header where the dependency ID should be added when tracking HTTP dependencies");
```

Comment:
```suggestion
                    throw new ArgumentException("Requires a non-blank value for the HTTP request header where the dependency ID should be added when tracking HTTP dependencies", nameof(value));
```

Reply:
That's a difficult one.  `NotNullOrWhitespace` throws an `ArgumentNullException` if the arg is null, otherwise an `ArgumentException`.

Reply:
We can start using that once we stop .NET Standard/.NET 6; for now, it's the same functionality as with the guard library.